lib/Parser.js

```diff
@@ -297,6 +297,26 @@ Parser.prototype.initializeEvaluating = function() {
 			}
 			return new BasicEvaluatedExpression().setString(result).setRange(expr.range);
 		});
+		this.plugin("evaluate TemplateLiteral", function(node) {
+			var i, quasis = node.quasis, exprs = [];
+			for (i = 0; i < quasis.length; i++) {
+				exprs.push(new BasicEvaluatedExpression().setString(quasis[i].value.cooked).setRange(quasis[i].range));
+				// possible improvement: evaluate node.expressions[i-1]
+				// and append to previous quasis if has a set value
+			}
+			if (exprs.length == 1) {
+				return exprs[0].setRange(node.range);
+			}
+			return new BasicEvaluatedExpression().setTemplateString(exprs).setRange(node.range);
+		});
+		this.plugin("evaluate TaggedTemplateExpression", function(node) {
+			if (this.evaluateExpression(node.tag).identifier !== 'String.raw') return;
+			var i, quasis = node.quasi.quasis, exprs = [];
+			for (i = 0; i < quasis.length; i++) {
+				exprs.push(new BasicEvaluatedExpression().setString(quasis[i].value.raw).setRange(quasis[i].range));
+			}
+			return new BasicEvaluatedExpression().setTemplateString(exprs).setRange(node.range);
```

Comment:
Is it safe to also do the `exprs.length == 1` simplification here?

The contents of the string itself will be correct, but if there is any code that assumes that the string's contents starts at `expr.range[0]+1` it will have a bad time.